### PR TITLE
Lazy load Brave Crypto Wallets

### PR DESCRIPTION
**Pull request by bbondy**
Fix https://github.com/brave/brave-browser/issues/9757
Fix https://github.com/brave/brave-browser/issues/8121
Fix https://github.com/brave/brave-browser/issues/6595
Fix https://github.com/brave/brave-browser/issues/8860
Fix https://github.com/brave/brave-browser/issues/8925
Fix https://github.com/brave/brave-browser/issues/8120

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

See first issue above.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: app/brave_generated_resources.grd [hunk 1]
```diff
@@ -726,18 +729,28 @@ By installing this extension, you are agreeing to the Google Widevine Terms of U
...
+        Setup Crypto Wallets
       </message>
       <message name="IDS_BRAVE_CRYPTO_WALLETS_USE_METAMASK" desc="Use MetaMask and don't use Crypto Wallets">
         Use MetaMask
       </message>
       <message name="IDS_BRAVE_CRYPTO_WALLETS_SETUP" desc="Use Crypto Wallets">
         SETUP
       </message>
+       <message name="IDS_BRAVE_CRYPTO_WALLETS_START_AND_RELOAD" desc="Start Crypto Wallets and reload">
+        Start Crypto Wallets and relaod
```
Comment (ryanml) [inline app/brave_generated_resources.grd:748]: small: `relaod` -> `reload`

File: components/brave_wallet/browser/brave_wallet_service.cc [hunk 2]
```diff
@@ -289,3 +291,40 @@ void BraveWalletService::OnExtensionLoaded(
     const extensions::Extension* extension) {
   RemoveUnusedWeb3ProviderContentScripts();
 }
+
+bool BraveWalletService::IsCryptoWalletsSetup() const {
+  PrefService* prefs = user_prefs::UserPrefs::Get(context_);
+  return prefs->HasPrefPath(kBraveWalletAES256GCMSivNonce) &&
```
Comment (ryanml) [inline components/brave_wallet/browser/brave_wallet_service.cc:297]: I'm not familiar with `HasPrefPath`, will this just check if the preference exists at all on the profile, or if it is non-empty? I think it's the former it should be an non-empty check.

File: app/brave_generated_resources.grd (hunk 1 above)
Comment (bbondy) [reply app/brave_generated_resources.grd:748]: fixed

File: components/brave_wallet/browser/brave_wallet_service.cc (hunk 2 above)
Comment (bbondy) [reply components/brave_wallet/browser/brave_wallet_service.cc:297]: I think it's correct:
```
  // Returns true if a value has been set for the specified path.
  // NOTE: this is NOT the same as FindPreference. In particular
  // FindPreference returns whether RegisterXXX has been invoked, where as
  // this checks if a value exists for the path.
  bool HasPrefPath(const std::string& path) const;
```

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.11.25 Chromium: 81.0.4044.138 (Official Build) nightly (64-bit)
-- | --
Revision | 8c6c7ba89cc9453625af54f11fd83179e23450fa-refs/branch-heads/4044@{#999}
OS | Linux

Test for #9757

Scenario 1: 
- Verified on a fresh profile there is no process started until opted in. 
- Verified closing the browser and reopening doesn't list Crypto Wallet process
- Verified closing and reopening browser and visiting `brave://wallet` start the process 
- Verified with process running, visiting Crypto Kitties doesn't show info bar and `window.web3` is available in the tab
- Verified closing and relaunching browser and visiting Crypto Kitties shows info bar to start CW and reload the tab
- Verified setting process to always load works between relaunches

Scenario 2:
- Verified on a clean profile visiting CryptoKittes brings up info bar  to setup Crypto
- Verified clicking on setup in info bar loads `brave://wallet` in new tab and shows opt-in page
- Verified after opt-in reloading the page provides `window.web3` on the tab

Scenario 3: 
- Verified installing MM and setting it as default doesn't start the process. 
- Verified visiting CryptoKittes and setting MM as default also sets in setting 
- Close and relaunch browser doesn't start the process
- Verified opting to CW doesn't change settings for default web3 provider

Scenario 4: 
- Verified installing MM on a clean profile and visiting CryptoKitties shows info bar to choose default Dapp provider. Selecting CW sets its as default Web3 provider in settings
- Verified closing and relaunching the browser and visiting CryptoKitties shows info bar asking to restart the CW process

Scenario 5: 
- Verified from an existing profile where CW is fully setup, upgrading shows no process for CW in task manager
- Verified visiting `brave://wallet` loads the extension and works fine

Scenario 6:
- Verified from an existing profile where CW is loaded but not setup, upgrading shows no process for CW in task manager
- Verified visiting `brave://wallet` shows opt-in page

Ran into #9821 for scenarios 5/6 which is fixed in https://github.com/brave/brave-core/pull/5591


-------
Test for #8121

- Verified on a clean profile visiting CryptoKitties shows info bar and selecting CW loads `brave://wallet` in a new tab and goes into opt-in state
------

Test for #6595

- Verified having a window size of 400px by 400px and visiting `brave://wallet` loads the opt-in webUI which is responsive and scrollable

------

Test for #8925

Verified there is no constant polling in the background when the process is not running 

-------

Test for #8860

- Verified there is no Infura calls when the process is not running
-------

Test for #8120

- Verified visiting a Dapp automatically triggers info bar to setup CryptoWallet
- Verified if MM is installed and visiting a Dapp automatically triggers info bar to choose between CW and MM as default Dapp provider

**Pull request merged**